src/JBrowse/Model/XHRBlob.js

```diff
@@ -141,19 +153,34 @@ var XHRBlob = declare( FileBlob,
 
     fetch( callback, failCallback ) {
         const length = this.end === undefined ? undefined : this.end - this.start + 1
-        if (length < 0) debugger
-        globalCache.getRange(this.url, this.start, length)
-            .then(
-                this._getResponseArrayBuffer.bind(this,callback),
-                failCallback,
-            )
+        if (length < 0) {
+            throw new Error('Length less than 0 received')
+        } else if(length == undefined && this.start == 0) {
+            getfetch(this.url).then(response => {
+                response.arrayBuffer().then(res => {
+                    callback(res)
+                }, failCallback)
+            }, failCallback)
+        } else {
```

Comment:
This could be much more nicely written like:

```
getfetch(this.url)
.then(res => res.arrayBuffer())
.then(res => callback)
.catch(failCallback)
```